src/test/java/org/prebid/server/bidder/madsense/MadsenseBidderTest.java

```diff
@@ -0,0 +1,399 @@
+                        .ext(givenImpExt("testCompanyId")),
+                builder -> builder.id("videoImp2").video(Video.builder().build())
+                        .ext(givenImpExt("otherCompanyId")));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = target.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1);
+        assertThat(result.getValue().getFirst().getImpIds()).containsExactlyInAnyOrder("videoImp1", "videoImp2");
+        assertThat(result.getValue().getFirst().getPayload().getImp())
+                .extracting(Imp::getId)
+                .containsExactlyInAnyOrder("videoImp1", "videoImp2");
+        assertThat(result.getValue().getFirst().getUri())
+                .isEqualTo(ENDPOINT_URL + "?company_id=testCompanyId");
+    }
+
+    @Test
+    public void makeHttpRequestsShouldHandleMixedBannerAndVideoImps() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(
+                builder -> builder.id("bannerImp1").banner(Banner.builder().build())
+                        .ext(givenImpExt("testCompanyId")),
+                builder -> builder.id("videoImp1").video(Video.builder().build())
+                        .ext(givenImpExt("otherCompanyId1")),
+                builder -> builder.id("bannerImp2").banner(Banner.builder().build())
+                        .ext(givenImpExt("otherCompanyId2")),
+                builder -> builder.id("videoImp2").video(Video.builder().build())
+                        .ext(givenImpExt("otherCompanyId3")));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = target.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(3);
+
+        assertThat(result.getValue())
+                .filteredOn(req -> req.getPayload().getImp().getFirst().getBanner() != null)
+                .hasSize(2)
+                .extracting(
+                        HttpRequest::getImpIds,
+                        request -> request.getPayload().getImp().getFirst().getId())
+                .containsExactlyInAnyOrder(
+                        tuple(Set.of("bannerImp1"), "bannerImp1"),
+                        tuple(Set.of("bannerImp2"), "bannerImp2"));
+
+        assertThat(result.getValue())
+                .filteredOn(req -> req.getPayload().getImp().getFirst().getVideo() != null)
+                .hasSize(1)
+                .first()
+                .satisfies(videoReq -> {
+                    assertThat(videoReq.getImpIds()).containsExactlyInAnyOrder("videoImp1", "videoImp2");
+                    assertThat(videoReq.getPayload().getImp())
+                            .extracting(Imp::getId)
+                            .containsExactlyInAnyOrder("videoImp1", "videoImp2");
+                });
```

Comment:
```
        assertThat(result.getValue())
                 .extracting(HttpRequest::getImpIds)
                 .containsExactly(
                         Set.of("bannerImp1"),
                         Set.of("bannerImp2"),
                         Set.of("videoImp1", "videoImp2"));
```

Reply:
If you have problems with set equality, you can add a custom comparator or use `anySatisfy` 3 times.